go/libraries/doltcore/remotestorage/chunk_store.go

```diff
@@ -135,22 +137,19 @@ func (dcs *DoltChunkStore) getRepoId() *remotesapi.RepoId {
 // Get the Chunk for the value of the hash in the store. If the hash is absent from the store EmptyChunk is returned.
 func (dcs *DoltChunkStore) Get(ctx context.Context, h hash.Hash) (chunks.Chunk, error) {
 	hashes := hash.HashSet{h: struct{}{}}
-	foundChan := make(chan *chunks.Chunk, 1)
-	err := dcs.GetMany(ctx, hashes, foundChan)
-
+	var found *chunks.Chunk
+	err := dcs.GetMany(ctx, hashes, func(c *chunks.Chunk) { found = c })
 	if err != nil {
 		return chunks.EmptyChunk, err
 	}
-
-	select {
-	case ch := <-foundChan:
-		return *ch, nil
-	default:
+	if found != nil {
+		return *found, nil
+	} else {
 		return chunks.EmptyChunk, nil
 	}
 }
 
-func (dcs *DoltChunkStore) GetMany(ctx context.Context, hashes hash.HashSet, foundChunks chan<- *chunks.Chunk) error {
+func (dcs *DoltChunkStore) GetMany(ctx context.Context, hashes hash.HashSet, found func(*chunks.Chunk)) error {
 	ae := atomicerr.New()
 	wg := &sync.WaitGroup{}
 	foundCmp := make(chan nbs.CompressedChunk, 1024)
```

Comment:
Thoughts on pushing this to the caller and allowing them to setup and use (on not) a channel in their callback as needed.  In cases where a channel isn't needed from a perf perspective?

Reply:
I guess by kicking off a go routine here and using the channel it means that we don't have to do that everywhere this is called.

Reply:
Yeah, not using the channel is definitely nicer:

https://github.com/dolthub/dolt/pull/938/commits/89780ffd88a2dbd1b95625bed036e38e7b198104

Still partially highlights the subtlety around threading though. These interfaces don't take the `*errgroup.Group`, so maybe calls to the callback should be serialized.